File: core/src/com/mygdx/xcube/block/HollowBar.java
package com.mygdx.xcube.block;

import static com.badlogic.gdx.math.MathUtils.round;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.Sprite;
import com.badlogic.gdx.math.Rectangle;

public class HollowBar extends TerrainBlock {

    public HollowBar(boolean rotation, int x, int y){
        super(x,y);
        this.sprite = setSprite("greybarv2.png");

        dx=round(this.getSprite().getWidth());
        dy=round(this.getSprite().getHeight());
        rectangle = new Rectangle(x-dx*2, y+dy/4, dx*5, dy/2);
        if (rotation) {
            rectangle = new Rectangle(x+dy/4, y-dx*2, dy/2, dx*5);
            dx = dy;
            dy = round(this.getSprite().getWidth());
        }
        isSquare=false;
        this.spritebleu = setSprite("V2/bluebar1.png");
        this.spritebleu2 = setSprite("V2/bluebar2.png");
        this.spriterouge = setSprite("V2/redbar1.png");
        this.spriterouge2 = setSprite("V2/redbar2.png");

    }

}

File: core/src/com/mygdx/xcube/block/HollowBar.kt
package com.mygdx.xcube.block

import com.badlogic.gdx.math.Rectangle

class HollowBar(rotation: Float, x: Float, y: Float) : Block(x, y) {
    var rotation: Float

    init {
        dx = 500f
        dy = 96f
        rectangle = Rectangle(x, y, dx, dy)
        //La rotation semble assez inutile, il faudra la virer plus tard. Quand tu dessines un
        //sprite avec une certaine rotation (méthode batch.draw()), ça ne modifie pas la hitbox
        //du rectangle. Du coup, on a le sprite qui est tourné de 90° par rapport à la hitbox
        //qui n'a pas bougé. Et je n'ai pas trouvé de paramètre "rotation" pour le rectangle.
        //A la place j'invers la largeur et la longueur du rectangle en fonction de la rotation
        //que l'on met en argument
        this.rotation = rotation
        if (rotation == 90f) {
            rectangle!!.setSize(96f, 500f)
            dx = 96f
            dy = 500f
        }
        isFree = true
    } //public HollowBar[] AddNeighbor(HollowBar[] bars) {
    //    return bars.add(this);
    //}
}